core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -2073,6 +2073,37 @@ package object config {
       .timeConf(TimeUnit.MILLISECONDS)
       .createOptional
 
+  private[spark] val SPECULATION_EFFICIENCY_ENABLE =
+    ConfigBuilder("spark.speculation.efficiency.enabled")
+      .doc("When set to true, spark will evaluate the efficiency of task processing through the " +
+        "stage task metrics and only need to speculate the inefficient tasks. A task is " +
+        "inefficient when its data process rate is less than the average data process " +
+        "rate of all successful tasks in the stage multiplied by a multiplier.")
+      .version("3.4.0")
+      .booleanConf
+      .createWithDefault(true)
+
+  private[spark] val SPECULATION_EFFICIENCY_TASK_PROCESS_MULTIPLIER =
+    ConfigBuilder("spark.speculation.efficiency.processMultiplier")
+      .doc("A multiplier for evaluating the efficiency of task processing. A task is inefficient " +
+        "when its data process rate is less than the average data process rate of all " +
+        "successful tasks in the stage multiplied by the multiplier.")
+      .version("3.4.0")
+      .doubleConf
+      .checkValue(v => v > 0.0 && v <= 1.0, "multiplier must be in (0.0, 1.0]")
+      .createWithDefault(0.75)
+
+  private[spark] val SPECULATION_EFFICIENCY_TASK_DURATION_FACTOR =
+    ConfigBuilder("spark.speculation.efficiency.durationFactor")
+      .doc(s"When a task duration is large than the factor multiplied by the threshold which " +
+        s"may be ${SPECULATION_MULTIPLIER.key} * successfulTaskDurations.median or " +
+        s"${SPECULATION_MIN_THRESHOLD.key}, and it should be considered for " +
+        s"speculation to avoid that it is too late to launch a necessary speculation.")
```

Comment:
"A task will be speculated anyway as long as its duration has exceeded the value of multiplying the factor and the time threshold (either be {SPECULATION_MULTIPLIER.key} \* successfulTaskDurations.median or {SPECULATION_MIN_THRESHOLD.key})" regardless of it's efficient or not. This avoids missing the tasks when  task slow isn't due to data process rate."